File: back/src/main/java/com/back/domain/report/report/repository/ReportedMessageRepository.java
package com.back.domain.report.report.repository;

import com.back.domain.report.report.entity.ReportedMessage;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface ReportedMessageRepository extends JpaRepository<ReportedMessage, Long> {
    List<ReportedMessage> findByReportIdOrderBySentAtAsc(Long reportId);

    Optional<ReportedMessage> findByUuid(UUID uuid);
}

File: back/src/main/kotlin/com/back/domain/report/report/repository/ReportedMessageRepository.kt
package com.back.domain.report.report.repository

import com.back.domain.report.report.entity.ReportedMessage
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface ReportedMessageRepository : JpaRepository<ReportedMessage, Long> {
    fun findByReportIdOrderBySentAtAsc(reportId: Long): List<ReportedMessage>

    fun findByUuid(uuid: UUID): ReportedMessage?
}
